# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/omar-khaled-opal/flyrank-intern/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

**Research question:** Which content items should be prioritized for review or refresh based on their observed freshness and search-performance signals?

**Decision:** Decide which content items are more worth prioritizing for a content review or refresh.

**Action:** A content or SEO team could review the prioritized items and decide whether to update, improve, or leave them unchanged.

**Cost of a wrong recommendation:** Prioritizing the wrong content could waste limited content and SEO resources. Missing a useful refresh opportunity could also mean leaving potential search performance improvements unexplored. Therefore, any result should be treated as decision support rather than a guarantee that refreshing a page will improve its performance.


In [3]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Average CTR:", df["ctr"].mean())
print("Average position:", df["avg_position"].mean())
print("Average content age:", df["content_age_days"].mean())

Rows: 30000
Average CTR: 0.5107333333333334
Average position: 16.342380000000002
Average content age: 256.1678


The starter dataset contains 30,000 content items. The measured average CTR is about 0.51%, while the average search position is about 16.34. The average content age is about 256 days. These numbers show that the dataset contains a substantial number of content observations together with measurable search-performance and freshness signals, making it reasonable to investigate whether these signals can support content-refresh prioritization.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.